.github/workflows/ci.yml

```diff
@@ -0,0 +1,179 @@
+        # workaround: https://github.com/actions/virtual-environments/issues/1536#issuecomment-698537248
+        run: |
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key 2>/dev/null | sudo apt-key add -
+          sudo add-apt-repository 'deb http://apt.llvm.org/focal llvm-toolchain-focal-10 main' -y
+          sudo apt update
+          sudo apt install kcov g++-10 valgrind libcurl4 libcurl4-openssl-dev
+          sudo apt install clang-10 clang-tools llvm
+
+      - name: Installing Linux 32-bit Dependencies
+        if: ${{ runner.os == 'Linux' && env['TEST_X86'] }}
+        run: sudo apt update && sudo apt install gcc-multilib g++-multilib
+
+      - name: Installing CodeChecker
+        if: ${{ contains(env['RUN_ANALYZER'], 'code-checker') }}
+        run: |
+          sudo apt install clang-tidy curl doxygen gcc-multilib python3-dev python3-virtualenv
+          git clone https://github.com/Ericsson/CodeChecker.git --depth 1
+          cd CodeChecker
+          BUILD_LOGGER_64_BIT_ONLY=YES make standalone_package
+          echo "::add-path::$PWD/build/CodeChecker/bin"
+
+      - name: Expose llvm PATH for Mac
+        if: ${{ runner.os == 'macOS' }}
+        run: echo "::add-path::/usr/local/opt/llvm/bin/"
+
+      - name: Installing Android SDK Dependencies
+        if: ${{ env['ANDROID_API'] }}
+        run: |
+          export ANDROID_IMAGE="system-images;android-$ANDROID_API;google_apis;x86"
+          echo "Downloading ndk;$ANDROID_NDK and $ANDROID_IMAGE"
+          echo "y" | $ANDROID_HOME/tools/bin/sdkmanager --install \
+            "ndk;$ANDROID_NDK" "$ANDROID_IMAGE" | \
+            grep -v "\[=" || true # suppress the progress bar, so we get meaningful logs
+
+      - name: Starting Android Simulator
+        if: ${{ env['ANDROID_API'] }}
+        run: bash scripts/start-android.sh
+
+      - name: Test
+        shell: bash
+        run: |
+          pip install --upgrade --requirement tests/requirements.txt
+          [ "${{ matrix.CC }}" ] && export CC="${{ matrix.CC }}"
+          [ "${{ matrix.CXX }}" ] && export CXX="${{ matrix.CXX }}"
+          pytest --capture=no --verbose tests
+
+      - name: "Upload to codecov.io"
+        if: ${{ contains(env['RUN_ANALYZER'], 'cov') }}
+        uses: codecov/codecov-action@v1
+        with:
+          directory: coverage
+
+  archive:
+    name: Create Release Archive
+    runs-on: ubuntu-latest
+    needs: [lint, test]
+    if: ${{ needs.test.result == 'success' && github.repository_owner == 'getsentry' }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Remove unneeded files
+        shell: bash
+        run: rm -rf build .c* .e* .git* scripts Makefile external/breakpad/src/tools external/breakpad/src/processor
+
+      - name: Create source archive
+        shell: python
+        run: |
+          import shutil
+          shutil.make_archive("sentry-native-source", "zip", ".")
```

Comment:
fwiw, `zip` is installed on GitHub runners.

Reply:
hm… right, we don’t need this to be cross-platform, but its only running on linux…
:-D I kind of forget that for CI… anyhow, I think its fine for now, I don’t really want to waste more time on it…